### ISSUE TITLE
self-intersecting d, e, g mis-rendered

### ISSUE THREAD

**Issue by kberry** [silnrsi/font-gentium#36]
Hi Victor or anyone - 
In the prior release of gentium (6.101, I believe), it seems the definitions for the d, e, and glyphs (at least) were changed to have self-intersecting paths. We're guessing this was intentional, and we have not had any reports of problems with people using your original TrueType fonts. 

However, the Type 1 fonts created for/by TUG (with FontForge) faithfully reproduce those definitions, and it seems old Acrobat or other renderers mess up, creating white spaces in the middle of the glyphs. I'll attach a screenshot. This also affects some RIP renderers in commercial printers; one person reported having to change the font for his book because of this rendering bug.

So, our questions are:
1) do you want to change the definitions to avoid the self-intersections in the TrueType originals?
2) if not, are you ok with us modifying the Type 1 definitions to avoid them?

Please advise. Thanks. -karl

P.S. You don't need to read them, but for the record, here are the reports we received on our TUG mailing list for gentium:
https://tug.org/pipermail/gentium/2023q1/000039.html
https://tug.org/pipermail/gentium/2024q4/000066.html

![gentium-bug](https://github.com/user-attachments/assets/4b0c4802-4cea-4a38-9049-d7b44dc29140)
![ff-Screenshot_20241211_140349](https://github.com/user-attachments/assets/c4551b67-e233-439f-bca4-650d6b6bf8fa)

**Comment by jvgaultney** [comment]
This was a change made back in 2021 for v6.000. These glyphs always had intersecting paths in the source, but prior to v6 we removed the overlap in the build. You'll need to do the same in your Type 1 conversion process.

You're welcome to do that within our SIL-TUG RFN agreement, but you'll need to proof the fonts carefully. The overlap removal process is not always predictable, and each program has it's own method of doing it. You'll need to visually inspect every glyph in every font to verify that the overlap has been removed properly. I know of no program that does this 100% correctly. You may then need to manually fix up some glyphs.

**Comment by kberry** [comment]
Thanks for the info, discouraging though it is. I have my doubts that any volunteer will come forward to proof "every glyph in every font", or even to figure out the overlap removal. All we ever did in the past is run the conversion in FontForge. So perhaps the Type 1's for Gentium are at an end. I'll ask around.

I'm sure you have good reasons, but I can't help but be curious as to what they are -- why did you stop removing the overlap in your build? --thanks, karl.

**Comment by jvgaultney** [comment]
There were a few reasons why we stopped removing overlap:

- TrueType rasterizers were rarely showing dropouts, so there was little need. There were a handful of troublesome shapes that consistently had problems - usually where a bar crossed over a counter, as in the barred-o. We manually decomposed and removed the overlap permanently for those in the source.
- Being thorough with removing overlap would have required decomposing all composite characters which would have massively increased the font file sizes with little real benefit.
- We couldn't trust the remove overlap in the build to behave consistently over time, so we would have had to constantly reinspect every glyph in every font on any build we release, which in our full set of version 7 fonts (Gentium, Charis, Andika, Doulos SIL) has grown to over 111,000 glyphs.
- Overlap is actually a _good_ thing to have in variable fonts, and if we ever decide to produce a Gentium variable font we will likely want to rebuild many glyphs to add more more overlaps.

I do sympathize with how difficult it would be to remove overlap on the Type 1 fonts. Given that, and the upcoming expansion of the family to many more weights, it may be time to end their further development. In general, Type 1 is a relic of the past and has been no longer supported in the industry for a while now.

One of the benefits of the TUG work was the additional kerning you added, however the upcoming version 7 will have even more extensive kerning that goes beyond what the current Type 1 fonts have. I'll be the first to admit that the version 7 Cyrillic kerning, for example, might not be as perfectly finely-tuned as what others have done, but it supports a much wider range of languages other than Russian.

I hope that helps you understand our purposes and directions.
